pkg/platform/baremetal/baremetal.go

```diff
@@ -110,6 +111,22 @@ func (c *config) Initialize(ex *terraform.Executor) error {
 		return err
 	}
 
+	// Extract control plane chart files to cluster assets directory.
```

Comment:
Also, the code is horribly duplicated between all platforms, we should greatly avoid adding more duplication.

Reply:
Horribly? :scream:

Reply:
Yes, I'm just very annoyed by number of duplication and boilerplate in providers code, both Go and Terraform. That leads to bugs :(

Reply:
I'm not a fan of duplication either and do my best to eliminate it when there's an obvious way to do so.

Reply:
@invidian I've looked some more into the duplication issue you've brought up and I couldn't figure out an easy way to avoid it without doing significant refactors which greatly increase the scope of this PR. As it's fairly big as it is already, I suggest we solve the duplication problem and do the necessary refactoring in a separate PR (which we can work on right away as far as I'm concerned). However, I wouldn't like to keep delaying fixing the diff bug just because we add a small amount of duplication. WDYT?

Reply:
Sounds like a reasonable suggestion :+1: I'll have a look if there is an easy way to deduplicate it and if I figure it out, I'll make some proposal. If not, I'm fine with it as it is.